test/contrib/test_socks.py

```diff
@@ -87,6 +90,24 @@ def _address_from_socket(sock):
         raise RuntimeError("Unexpected addr type: %r" % addr_type)
 
 
+def _set_up_fake_getaddrinfo(monkeypatch):
```

Comment:
Thoughts on making this a fixture so `monkeypatch` is passed automatically? Then we can do `@pytest.mark.usefixtures("mock_getaddrinfo")`

Reply:
I think that explicit is better than implicit here: it's apparent when reading the test thank monkeypatching is going on. I prefer reserving `usefixtures` magic to situations where it's required. (But good call on noting that it's possible, I hadn't realized it!)

Now that you know my thoughts, please tell me if I should still do this changes, I'll be happy to.

Reply:
This is fine, my only hangup was passing a pytest fixture as a parameter when it could instead be setup within a fixture that depends on the `monkeypatch` fixture.